tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/V3/XUnitTestAssemblyRunnerRunV3Integration.cs

```diff
@@ -0,0 +1,94 @@
+// <copyright file="XUnitTestAssemblyRunnerRunV3Integration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.ComponentModel;
+using System.Reflection;
+using System.Threading.Tasks;
+using Datadog.Trace.Ci;
+using Datadog.Trace.Ci.Tags;
+using Datadog.Trace.ClrProfiler.CallTarget;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.XUnit.V3;
+
+/// <summary>
+/// Xunit.v3.TestAssemblyRunner`4.Run calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyName = "xunit.v3.core",
+    TypeName = "Xunit.v3.TestAssemblyRunner`4",
+    MethodName = "Run",
+    ParameterTypeNames = ["_"],
+    ReturnTypeName = "System.Threading.Tasks.ValueTask`1[Xunit.v3.RunSummary]",
+    MinimumVersion = "1.0.0",
+    MaximumVersion = "1.*.*",
+    IntegrationName = XUnitIntegration.IntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class XUnitTestAssemblyRunnerRunV3Integration
+{
+    internal static CallTargetState OnMethodBegin<TTarget, TContext>(TTarget instance, TContext context)
+        where TContext : ITestAssemblyRunnerContextV3
+    {
+        if (!XUnitIntegration.IsEnabled || instance is null)
```

Comment:
I assume `instance` can't be null because it's an instance method, should this be checking the context instance?
```suggestion
        if (!XUnitIntegration.IsEnabled || context.Instance is null)
```

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/6573/commits/c5f706fbd5d4b72fcfb9d21f779cc09c96c2bd87